### PR TITLE
core: ping: Add REST API

### LINKED ISSUE #652: core: frontend: Add information about connected sensor such as Ping1D or Ping360

### PR DESCRIPTION
**Pull request by patrickelectric**
- Fix #652 
- docker: patrickelectric/companion-core:ping_rest

### CODE REVIEW

**Review by Williangalvani — CHANGES_REQUESTED** [review]

File: core/services/ping/main.py [hunk 1]
```diff
@@ -2,32 +2,75 @@
...
+from typedefs import PingDeviceDescriptorModel
+from uvicorn import Config, Server
 
 from pingmanager import PingManager
 from pingprober import PingProber
 from portwatcher import PortWatcher
 
+app = FastAPI(
+    title="ArduPilot Manager API",
+    description="ArduPilot Manager is responsible for managing ArduPilot devices connected to Companion.",
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply core/services/ping/main.py]: Fixed.

File: core/services/ping/main.py [hunk 2]
```diff
@@ -2,32 +2,75 @@
...
+from loguru import logger
+from typedefs import PingDeviceDescriptorModel
+from uvicorn import Config, Server
 
 from pingmanager import PingManager
 from pingprober import PingProber
 from portwatcher import PortWatcher
 
+app = FastAPI(
+    title="ArduPilot Manager API",
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply core/services/ping/main.py]: Fixed.

File: core/services/ping/main.py [hunk 3]
```diff
@@ -2,16 +2,56 @@
...
+logger.info("Starting Ping Service.")
+
+# TODO: move to singleton
+ping_manager = PingManager()
+
+
+@app.get("/sensors", response_model=List[PingDeviceDescriptorModel], summary="Current sensors detected.")
+@version(1, 0)
+def get_sensors() -> Any:
+    return [PingDeviceDescriptorModel.from_descriptor(device) for device in ping_manager.devices()]
```
Comment (rafaellehmkuhl) [inline core/services/ping/main.py:35]: Maybe we should log this devices list, to make debugging easier?

Comment (patrickelectric) [reply core/services/ping/main.py:37]: Done.

**Review by rafaellehmkuhl — APPROVED** [review]

**Pull request merged**